storage/local/storage.go

```diff
@@ -129,12 +129,13 @@ const (
 type syncStrategy func() bool
 
 type memorySeriesStorage struct {
-	// numChunksToPersist has to be aligned for atomic operations.
-	numChunksToPersist int64         // The number of chunks waiting for persistence.
-	maxChunksToPersist int           // If numChunksToPersist reaches this threshold, ingestion will be throttled.
-	rushed             bool          // Whether the storage is in rushed mode.
-	rushedMtx          sync.Mutex    // Protects entering and exiting rushed mode.
-	throttled          chan struct{} // This chan is sent to whenever NeedsThrottling() returns true (for logging).
+	// archiveHighWatermark and numChunksToPersist have to be aligned for atomic operations.
+	archiveHighWatermark model.Time    // No archived series has samples after this time.
+	numChunksToPersist   int64         // The number of chunks waiting for persistence.
```

Comment:
Did you find another source than "The first word in a global variable or in an allocated struct or slice can be relied upon to be 64-bit aligned." from https://golang.org/pkg/sync/atomic/? It doesn't say 100% specifically that the second field in a struct following a 64-bit value will also be aligned, though it'd be weird if the compiler implemented it differently. I guess it's reasonably safe to assume that it's aligned.